### PR TITLE
Enable cargo-spellcheck for blockchain/chain_sync

### LINKED ISSUE #1687: Enable `cargo-spellcheck` for `blockchain/chain_sync`
**Issue summary**

Make sure that `cargo spellcheck` can run without errors in `blockchain/chain_sync`. For details, see the tracking issue.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
-  Added async, incentivize, P2P, synchronizer, nonces to the dictionary
- Fixed the crate, so it could pass the spellcheck
- I've added async instead of fuller versions, as it became a more popular word. 

**Reference issue to close (if applicable)**

Closes  #1687

**Other information and links**

### CODE REVIEW

**Comment by lemmih** [comment]
Always write a summary of the changes in the PR. The summary is also a good place to describe the design decisions you've made. For example, you could write about why you added `async` to the dictionary instead of using `asynchronous` or `asynchronously`.

File: .config/forest.dic [hunk 1]
```diff
@@ -17,3 +17,8 @@ tuple
 mutex
 RPC
 parameterize/D
+P2P
+synchronizer
+nonces
```
Comment (lemmih) [inline .config/forest.dic:22]:
```suggestion
nonce/S
```

Comment (tyshko5) [reply .config/forest.dic:22]: this doesn't work for some reason

**Comment by tyshko5** [comment]
Added

File: .config/forest.dic [hunk 2]
```diff
@@ -17,3 +17,8 @@ tuple
...
+async
+incentivize
```
Comment (LesnyRumcajs) [inline .config/forest.dic:24]: I'm kind of puzzled why do we need to add this. It's a normal word, no? Perhaps we can use a better "general" dictionary? Same goes for `nonce`.

**Review by LesnyRumcajs — APPROVED** [review]

File: .config/forest.dic (hunk 2 above)
Comment (lemmih) [reply .config/forest.dic:24]: Yeah, the builtin dictionary only has 48k words. My up-to-date local dictionary has 79k words (including `nonce` and `incentivize`). The dictionary is 841KiBi. Probably worth it to include it in the repository.

Comment (LesnyRumcajs) [reply .config/forest.dic:24]: Let's do it! :muscle:

**Review by lemmih — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
@tyshko5 mergio?

**Comment by tyshko5** [comment]
> @tyshko5 mergio?

I guess. But I'm not sure if I can merge anything

**Pull request merged**